### DISCUSSION TITLE
Downloading binaries does not work

### DISCUSSION THREAD

**Discussion by karna41317** [prisma/prisma#18036] [Q&A]
I am on a corporate network, using proxy env variable to generate prisma client as follows, 

package.json: 
```
.....
"build": "npm-run-all prisma:generate build:app",
"build:app": "next build",
 "prisma:generate": "prisma generate --data-proxy",
 ......
```
Dockerfile:

```
.............
ARG PRISMA_GENERATE_DATAPROXY
ARG NO_PROXY
ARG HTTP_PROXY
ARG HTTPS_PROXY

ENV PRISMA_GENERATE_DATAPROXY TRUE
ENV NO_PROXY .xxxxx.se
ENV HTTP_PROXY http://xxx.xxxx.se
ENV HTTPS_PROXY https://xxxx.xxxx.se

RUN yarn install --frozen-lockfile

RUN echo "PRISMA_GENERATE_DATAPROXY=$PRISMA_GENERATE_DATAPROXY"
RUN echo "NO_PROXY=$NO_PROXY"
RUN echo "HTTP_PROXY=$HTTP_PROXY"
RUN echo "HTTPS_PROXY=$HTTPS_PROXY"

RUN yarn build
............
```

Error log: 

```


#17 [builder 11/12] RUN yarn build
#17 0.461 yarn run v1.22.19
#17 0.516 $ npm-run-all prisma:generate build:app
#17 1.186 $ prisma generate --data-proxy
#17 2.218 Error: request to https://binaries.prisma.sh/all_commits/aead147aa326ccb985dcfed5b065b4fdabd44b19/linux-musl-openssl-3.0.x/migration-engine.gz.sha256 failed, reason: connect ECONNREFUSED 172.21.203.229:443
#17 2.693 error Command failed with exit code 1.
#17 2.693 info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
#17 2.704 ERROR: "prisma:generate" exited with 1.
#17 2.721 error Command failed with exit code 1.
#17 2.721 info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
#17 ERROR: executor failed running [/bin/sh -c yarn build]: exit code: 1

#

````

I tried setting PRISMA_ENGINE_MIRROR=http://binaries.prisma.sh but no luck. but same [url](https://binaries.prisma.sh/all_commits/aead147aa326ccb985dcfed5b065b4fdabd44b19/linux-musl-openssl-3.0.x/migration-engine.gz.sha256) works in browser

**Comment by nurul3101** [comment]
Hello @karna41317 👋 

Are you able to download the binaries outside your corporate network? Is download restricted in your network?

**Reply by karna41317** [reply]
I was able to download from browser. It downloaded a file with name: migration-engine.gz.sha256

**Reply by karna41317** [reply]
I tried generating client for following binary target config, 

```
generator client {
    provider      = "prisma-client-js"
    output        = "./../src/services/prisma-client"
    binaryTargets = ["native", "linux-musl-openssl-3.0.x"]
}
```

and it gave me following error

```
Environment variables loaded from .env
Prisma schema loaded from prisma/schema.prisma
> Downloading Prisma engines for Node-API for darwin and linux-musl-openssl-3.0.x [                    ] 0%Error: request to https://binaries.prisma.sh/all_commits/aead147aa326ccb985dcfed5b065b4fdabd44b19/linux-musl-openssl-3.0.x/libquery_engine.so.node.gz.sha256 failed, reason: getaddrinfo ENOTFOUND binaries.prisma.sh

```

**Reply by karna41317** [reply]
Also, I have removed all Proxy env variables, still got same error in docker logs while generating client

**Reply by nurul3101** [reply]
Can you try changing your network and see if Prisma engines still doesn't download?

**Comment by janpio** [comment]
What are you trying to achieve - make Prisma use some HTTP proxy that is allowed in your corporate environment, or make it generate a Prisma Data Platform Data Proxy Client, that does not use any of the engines and hence should not need to download them?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
